app/src/main/java/com/iic/shopingo/services/UserConnector.java

```diff
@@ -124,6 +99,38 @@ public void setCurrentUser(User currentUser) {
     editor.apply();
   }
 
+  void fetchUserDetails(Session session, final Task<User>.TaskCompletionSource taskCompletionSource) {
```

Comment:
This flow is a bit weird. It doesn't fetch the user's details, it logs in the user and sets the global `currentUser`.
The fact that it eventually also returns the user instance seems less important than the side-effect of login.